tests/e2e_test/utils_test.go

```diff
@@ -202,3 +291,67 @@ func GetKymaStateMetricCount(ctx context.Context, kymaName, state string) (int,
 
 	return 0, nil
 }
+
+func UpdateSampleCRSpec(ctx context.Context, name, namespace, resourceFilePath string, clnt client.Client) error {
+	sampleCR := &unstructured.Unstructured{Object: map[string]interface{}{
+		"apiVersion": "operator.kyma-project.io/v1alpha1",
+		"kind":       "Sample",
+	}}
+
+	if err := clnt.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, sampleCR); err != nil {
+		return err
+	}
+
+	if err := unstructured.SetNestedField(sampleCR.Object, resourceFilePath, "spec", "ResourceFilePath"); err != nil {
+		return err
+	}
+
+	return clnt.Update(ctx, sampleCR)
+}
+
+func CheckSampleCRIsInState(ctx context.Context, name, namespace string, clnt client.Client,
```

Comment:
the internal check state logic you can reuse for other resources, including those typed one, like manifestCR, not only SampleCR.

Then you can reuse it for `CheckManifestIsInState`, `CheckKymaIsInState`

Reply:
please also refactor and align this [IsKymaInState](https://github.com/kyma-project/lifecycle-manager/blob/174363eac89b6558c3393908fe2baaefff8bc17f/pkg/testutils/kyma.go#L111) with your `CheckKymaIsInState`

Reply:
Check Kyma will be refactored

Reply:
as mentioned [here](https://github.com/kyma-project/lifecycle-manager/pull/912#discussion_r1357156505), no matter which way you get manifest CR, then you already have manifest name, namespace, gvk, what prevents you reuse code?

basically, for this case, you can also reuse `CRIsInState` in `CheckManifestIsInState`